WIP: Ovirt_disk - Add ability to refresh passthrough disks

rhodain opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
A new parameter "host" is added. When this parameter is specified the newly created disk or an existing disk will refresh its information about the underlying storage via the specified host. This option is only valid for passthrough disks. This option requires at least the logical_unit.id to be specified.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ovirt_disk
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible 2.6.3
```

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
./hacking/test-module -m ./lib/ansible/modules/cloud/ovirt/ovirt_disk.py -a "name=MyDisk01 logical_unit=id=36001405fb6247061a564688a9baa34dba  host=RHV-H01"
<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

mkrizek committed:
include_vars: fix checking for a string in ignore_files

Commit:
Fixes HTTP redirect issue (#45513)

rhodain committed:
Ovirt_disk fails with "The 'value' parameter must be an integer"

Ovirt_disk failes with error:

An exception occurred during task execution. To see the full traceback, use
-vvv. The error was: TypeError: The 'value' parameter must be an integer
localhost | FAILED! => {
    "changed": false,
    "msg": "The 'value' parameter must be an integer"
}

when specifuing a port for iSCSI target for direct LUN creation. This
commit converts the port string into the integer.

rhodain committed:
Ovirt_disk fails with "The 'value' parameter must be an integer"

Ovirt_disk failes with error:

An exception occurred during task execution. To see the full traceback, use
-vvv. The error was: TypeError: The 'value' parameter must be an integer
localhost | FAILED! => {
    "changed": false,
    "msg": "The 'value' parameter must be an integer"
}

when specifuing a port for iSCSI target for direct LUN creation. This
commit converts the port string into the integer.

rhodain committed:
Ovirt_disk fails with "The 'value' parameter must be an integer"

Ovirt_disk failes with error:

An exception occurred during task execution. To see the full traceback, use
-vvv. The error was: TypeError: The 'value' parameter must be an integer
localhost | FAILED! => {
    "changed": false,
    "msg": "The 'value' parameter must be an integer"
}

when specifuing a port for iSCSI target for direct LUN creation. This
commit converts the port string into the integer.

rhodain committed:
Ovirt_disk - Add ability to refresh passthrough disks

A new paramter "host" is added. When this paramter is specified the newly
created disk or an existing disk will refresh its information about the
underlying storage via the specified host. This oprion is only valid for
passthrough disks. This option require at least the logical_unit.id to be
specified.

rhodain committed:
Ovirt_disk - Add ability to refresh passthrough disks

A new paramter "host" is added. When this paramter is specified the newly
created disk or an existing disk will refresh its information about the
underlying storage via the specified host. This oprion is only valid for
passthrough disks. This option require at least the logical_unit.id to be
specified.

rhodain committed:
Ovirt_disk - Add ability to refresh passthrough disks

A new paramter "host" is added. When this paramter is specified the newly
created disk or an existing disk will refresh its information about the
underlying storage via the specified host. This oprion is only valid for
passthrough disks. This option require at least the logical_unit.id to be
specified.

gundalow committed:
BOTMETA Validator + Bulk tidyup, support:(core,community,network) (#44903)

plugins/ is COMMUNITY
Set sensible defaults for directories
support:network for the platforms that we Networking SUPPORTS,
everything else is COMMUNITY
Mark other support:network (ansible-connection, etc)
Infoblox is support:core
contrib/ by definition should be support:community
Remove duplicated labels
Make yamllint happy(ier)
Adds sanity test to ensure BOTMETA.yml is valid

rhodain committed:
Formating fixes + minor index error fix

rhodain committed:
Formating fixes + minor index error fix

rhodain committed:
One more rormating fix

rhodain committed:
One more rormating fix

Deaton64 committed:
Update azure_rm_trafficmanagerprofile.py examples (#45649)

Wrong module name used in examples. changed `azure_rm_trafficmanager` to `azure_rm_trafficmanagerprofile`

mscherer committed:
[voss_facts] remove uneeded global declaration of variable warnings (#45653)

Since the variable is used only once, no need to declare it global.
Remove a warning from linter (lgtm.com)

sivel committed:
Flip default for AGNOSTIC_BECOME_PROMPT (#45563)

* Flip default for AGNOSTIC_BECOME_PROMPT

* simplify AGNOSTIC_BECOME_PROMPT porting guide entry

* clarify AGNOSTIC_BECOME_PROMPT changelog entry

* fixes version typo

* uses alternate wording (rather than escaping *)

maugli13 committed:
Return correct version on installed VyOS (#39115)

* Return correct version on installed VyOS

Previously existing regexp will shows only "VyOS" without numeric output of router version.
For example: from  "Version:      VyOS 1.1.6" only VyOS will be written in ansible_net_version variable
For more informative output numeric value should be returned as well

* Fixed unittests

ktdreyer committed:
add docstring for ansible.inventory.helpers.get_group_vars

Document the expected parameter type and return type of
get_group_vars().

mattclay committed:
Minor cleanup of code-smell tests. (#45658)

* Minor cleanup of code-smell tests.
* Add exception handling for YAML load.

lwm committed:
Linode: clarify how to create/delete linode machines with `linode_id`. (#45659)

* Attempt to explain `linode_id` a bit better.

Don't include in any example that creates a Linode.

Based on comments in

> https://github.com/ansible/ansible/issues/45403#issuecomment-419752856

* Add simple creation example. Show how to pass `linode_id`.

eisjcormier committed:
[aws_ssm_parameter_store] use describe_parameters paginator (#45632)

Fix the service reaching an internal limit while processing the results and returning unexpected data

Commit:
Lenovo doc update (#45483)

* Merge issue

* Update cnos_rollback.py

* Updating license for the refactored method

* Update cnos_rollback.py

* Removing the BSD License as suggested by Legal

* To add Documentation for ENOS as well as CNOS

* Merge issue

* Revert "To add Documentation for ENOS as well as CNOS"

This reverts commit 80e6e39054be0c3a8f95d16dc39ca9d93baf8c4b.

* Adding Docs for ENOS and CNOS

* Update cnos.py

* Update cnos_rollback.py

* Update cnos.py

* Update platform_cnos.rst

* Update platform_enos.rst

* Removed version 2.7

* Removing 2.7

julianladisch committed:
Suggest full fingerprint for apt_key. (#44697)

* Suggest full fingerprint for apt_key.
Background: https://gwolf.org/node/4070 (Gunnar Wolf: Stop it with those short PGP key IDs!)
I've put the full fingerprint into the examples.
* avoids the phrase 'best practice'

sivel committed:
Add new expand_shell argument for run_command, to disable expanding shellisms (#45620)

* Add new expand_shell argument for run_command, to disable expanding shellisms. Fixes #45418

* s/expand_shell/expand_user_and_vars/g

mscherer committed:
Do not reset permissions in checkmode

If using authorized_key on a directory with non standard permissions,
using checkmode will reset the permission silently.

dagwieers committed:
Add ACI dev_guide for $team_aci

resmo committed:
twilio: fix multi to_number support (#44878)

* twilio: fix multi to_number support

* use json helper

mscherer commented:
This PR would conflict with https://github.com/ansible/ansible/pull/45643 that you opened before :/

mscherer commented on lib/ansible/modules/cloud/ovirt/ovirt_disk.py:
```diff
@@ -578,12 +586,23 @@ def main():
+    # an existing disk if already available inthe environment.
+    if host and lun.get("id") is None:
+        module.fail_json(
+            msg="Can not use parameter host ({0!s}) withouti "
```
I guess the "i" should be removed, or exchanged for a "s" to go with "pecifying" after

rhodain replied:
Corrected

mscherer commented on lib/ansible/modules/cloud/ovirt/ovirt_disk.py:
```diff
@@ -693,6 +713,19 @@ def main():
                 ret = disk_attachments_module.remove()
 
+        # When host is specified and the disk is not being removed refresh the
+        # information abou thte LUN.
```
Nit picking but while on it, can you correct the comment ?

rhodain replied:
Corrected

mscherer commented on lib/ansible/modules/cloud/ovirt/ovirt_disk.py:
```diff
@@ -693,6 +713,19 @@ def main():
 
+        # When host is specified and the disk is not being removed refresh the
+        # information abou thte LUN.
+        host = module.params.get('host')
```
Duplicate from line 597

rhodain replied:
You are right, so I have removed this line.

mscherer commented on lib/ansible/modules/cloud/ovirt/ovirt_disk.py:
```diff
@@ -146,6 +146,14 @@
+        description:
+            - "When the hypervisor name is specified the newly created disk or
+               an existing disk will refresh its information about the
+               underlying storage. This oprion is only valid for passthrough
```
s/oprion/option/

rhodain replied:
Corrected

mscherer commented on lib/ansible/modules/cloud/ovirt/ovirt_disk.py:
```diff
@@ -146,6 +146,14 @@
                your playbook accordingly to not export the disk all the time.
                This option is valid only for template disks."
         version_added: "2.4"
+    host:
```
Could a more descriptive name be given ? I am not sure people would assume that giving the host will trigger a refresh (at least I don't, but I do not know ovirt well enough).

rhodain replied:
I added some additional information.

gundalow replied:
Is `host` the term that ovirt uses?
Just want to ensure their isn't confusion between what Ansible refers to as a host vs ovirt

mscherer replied:
Yeah, but so, could it be renamed, or does it make sense for ovirt users ? Again, i am sysadmin and I find that naming to be confusing (as I guess "host" is used for the host to connect usually) , but I am not using ovirt, so maybe it is obvious for people who are using it.

mwperina replied:
It's correct for oVirt/RHV, by host we mean a machine where VMs are running.

remyleone committed:
Add a netbox dynamic inventory plugin (#45347)

rhodain committed:
Fixing typos and documentation details

rhodain committed:
Fixing typos and documentation details

NilashishC committed:
Added github alias to team networking and vyos (#45711)

gundalow committed:
BOTMETA: Only notify sivel & gundalow on code changes to validate-modules (#45716)

webknjaz committed:
Fix collections.deque import in compat module

skylerbunny committed:
Add undocumented configuration parameter and explain in porting guide (#36059)

* Add undocumented configuration parameter and explain in porting guide

Qalthos committed:
There are other instances of transform being used... update them as well (#45663)

Commit:
Simplify PPA installation for Ubuntu (#45690)

Adds the `--update` flag to automatically update the cache and the `--yes` flag to disable the Ansible repo description and skips confirmation dialog. +label: docsite_pr

dpassante committed:
Cloudstack: fix support for some VPC service capabilities (#45727)

* Fix support for VPC capabilities such as redundant routers or region level VPC

* Add integration test cases for "region level VPC" and "distributed router" capabilities

gundalow committed:
Allow removed_in: 2.12 (#45732)

Qalthos committed:
These won't get upgraded to bool if ansible doesn't know they are (#45736)

Qalthos committed:
Add initial napalm connection plugin (#45224)

* Add initial napalm connection plugin

* Fix review comments

dagwieers committed:
reboot: Fix typo and support bare Linux systems (#45607)

* reboot: Fix typo and support bare Linux systems

This fixes a problem for bare Linux systems that do not support 'who -b' or 'uptime -s'.

* Accumulate stdout and stderr information

lingxiankong committed:
os_loadbalancer: support to create populated load balancer (#44619)

The user can use os_loadbalancer module to configure a fully functional
load balancer by specifying the sub-resources definition in the module,
rather than combining os_listener, os_pool and os_member modules.

s-hertel committed:
ec2_group: fix regression for targets that are a list containing strings and lists (#45594)

* Fix targets that may be a list containing strings and lists which worked prior to 2.6.

* Add ec2_group integration tests for lists of nested targets

* changelog

* Add diff mode support for lists of targets containing strings and lists.

s-hertel committed:
[iam_role] A hyphen with nothing following becomes [None] rather than []. Fix example syntax. (#45742)

s-hertel committed:
[aws] fix version_added for elb_classic_lbmodule (#45559)

afeld committed:
Link to the Galaxy platforms list from the meta file template (#34046)

* Link to the Galaxy platforms list from the meta file template
Fixes https://github.com/ansible/galaxy/issues/52.
* make ansible/ansible text match mazer text on platforms

Qalthos committed:
Update from napalm PR (#45743)

mscherer committed:
Add a alias from organisation to organization (#45688)

Since 'organization' is the spelling used accross all others modules,
I think it would be better to at least have this one as a alias.

Organisation is the UK/Australia/NZ spelling, while organization is the
US one.

remyj38 committed:
Add example for multiple package selection with dnf (#45610)

Thanks for enhancing the Ansible documentation!

Commit:
updated documentation in vmware_guest.py (#45481)

Added note regarding online disk resizing and changed wording of a few sentences. Also added vSphere 6.7 as "tested on" as it works very well.

zharalim committed:
Enhance the note about fileglob being local (#45398)

Made the note clearer and instructed to use find module for remote operations.
Relating to #34497

zwindler committed:
Wrong parameter in scaleway sshkey module example (#45737)

* Update scaleway_sshkey.py
fix: typo "present" in example

samccann committed:
Add 'Wait for' example to asa_command module  (#45190)

* Added example to register result of a ping test, including a wait_for command.
* added names to tasks
* updated examples to remove deprecated commands etc
* removed group_var

rwanyoike committed:
Add the key_name/value_name options to the dict2items filter (#45550)

* Add the key_name/value_name options to dict2items - as with items2dict, allow users to configure the key/value name for dict2items, add "version added" and examples

s-hertel committed:
[aws unit test utils] only create a recordings directory when the env var is set (#45752)

* Only create placebo recording test directories when the environment variable PLACEBO_RECORD is set

s-hertel committed:
Remove placeboify from unit tests that are not calling AWS (i.e. creating a recording) (#45754)

mscherer committed:
Add a default man path (#45689)

It seems that on some Linux distribution (Fedora 28, Debian), man will
not fallback on a default path if MANPATH is set. So using the env-setup
script will prevent man from working.

vmindru committed:
adding git+ssh uri scheme (#36025)

* incorporates git+ssh into existing example

bcoca committed:
allow openstack invetnory to work cacheless

fixes #45431

mscherer committed:
Port the aix_inittab to python 3 (#45678)

Since izip no longer exist on python 3, we need to use zip
Fix #45220

willthames committed:
Improve iam_group exception handling (#45599)

* Improve iam_group exception handling

Use AnsibleAWSModule for iam_group and handle BotoCoreErrors
as well as ClientErrors. Use fail_json_aws to improve error messages

* Add minimal iam_group test suite

Update some of the read-only IAM permissions (this is not sufficient
to run the test suite but it gets further than it did until it tries
to add a (non-existent) user)

* Clean up after tests

felixfontein committed:
Don't parse/interpret options if state is 'absent'. (#45700)

mattclay committed:
Fix unit tests which modify the source tree. (#45763)

* Fix CNOS unit test log usage.
* Use temp dir for Galaxy unit tests.
* Write to temp files in interfaces_file unit test.
* Fix log placement in netapp_e_ldap unit test.

mattclay committed:
Fix unit test parametrize order on Python 3.5.

dariko committed:
New module: add docker_stack module (clound/docker/docker_stack) (#24588)

* add docker_stack module + tests

mnecas committed:
ovirt add vnic (#44716)

* ovirt add vnic

mscherer commented on lib/ansible/modules/cloud/ovirt/ovirt_disk.py:
```diff
@@ -146,6 +146,16 @@
+            - "When the hypervisor name is specified the newly created disk or
+               an existing disk will refresh its information about the
+               underlying storage( Disk size, Serial, Product ID, Vendor ID ...)
+               The specified host will be used for garherting the storage
```
I think 'garherting' is not the spelling you want

mwperina replied:
Roman, you probably wanted to use 'gathering', right?

mscherer commented:
Also, the module say it support check_mode, but it does seems to be the case at all :/

machacekondra commented:
> Also, the module say it support check_mode, but it does seems to be the case at all :/

It does. The logic is mostly in the `module_utils/ovirt.py`. But there may be cases where it's not supported, but it's an issue and we need to fix it.

mrmagooey committed:
VMware: fix return to return a tuple instead of a bool (#45764)

fixes "'bool' object is not iterable" exception when `res` is falsey

s-hertel committed:
[ovirt] remove 2.8 deprecations (#45506)

sivel committed:
If including a file resulted in AnsibleFileNotFound, clarify error message. Fixes #45661 (#45667)

sivel committed:
Support transfering empty files to target host. Fixes #36725 (#45751)

samdoran committed:
Make sure strings are bytes for Py2/3 compatibility (#45746)

MrOwen committed:
Add workaround for non-standard kerberos environments (#41465)

* Add workaround for non-standard MIT kerberos environments

* Generalize platform specific troubleshooting steps for Kerberos

joaopsys committed:
Add eigrp as a possible l3protocol in aci_l3out

Also fixes an error related to the required parameters. The l3out parameter is required but only 'name' was being accepted, and that should only be the alias.

joaopsys committed:
Update aci_l3out documentation

mattclay committed:
Fix ansible-test unit test execution. (#45772)

* Fix ansible-test units requirements install.
* Run unit tests as unprivileged user under Docker.

hemskgren committed:
update asa_config with bgp doc example (#41376)

* update asa_config with bgp doc example

+label: docsite_pr
Issue ansible/community#311

* fix spelling

hemskgren committed:
asa_config add doc l3 interface (#41250)

* asa_config add doc l3 interface

+label: docsite_pr
Issue ansible/community#311

* fix interface idempotent

* newline, names, and debug:var

seymar committed:
Update influxdb_database.py (#39984)

InfluxDB >= 1.2.4 is not supported

+label: docsite_pr

joaopsys committed:
Update version_added on the asn parameter

mattclay committed:
Change group on docker_stack test.

This should provide a temporary fix for conflicts with the docker_swarm test.

s-hertel committed:
ec2_group - fix VPC precedence for security group targets (#45787)

Update the dictionary with the preferred values last to get the right order of VPC precedence

Fixes #45782

acozine committed:
Restore license agreement (#45809)

* adds page for CLA in community docs with links from dev and community docs

mattclay committed:
Run unit tests in parallel. (#45812)

mattclay committed:
Minor fixes for unit test delegation.

resmo committed:
vultr_server: fix diff for user data (#45753)

resmo committed:
vultr: fix for unreliable API behavior (#45712)

caphrim007 committed:
Update f5 module utils from downstream (#45819)

* various refactoring
* lgtm fixes
* bigiq support to different auth providers

caphrim007 committed:
Removes unused f5 imports (#45821)

ZuZuD committed:
EFS - add support for new Provisioned Throughput (#43253)

* efs.py: Add support for EFS provisioned throughput

* efs_facts.py: Add support for EFS provisioned throughput

* efs_facts integration tests updated with provision throughput

* efs_facts: Tests refactoring - add failure and success playbook according to botocore version.

* efs_facts: More tests and new option descriptions adjustment

* efs_facts tests renamed to efs

Nani-o committed:
Fix mixed output from ansible and lxd when using the lxd connection plugin (#45246)

* Replace fetch and put operations with Popen instead of call to prevent lxd to mess the Ansible output

* Remove extra blank line

mattclay committed:
Block network access for unit tests in docker.

caphrim007 committed:
Adds bigiq_device_facts module (#45822)

This module can be used to query a variety of facts from a BIG-IQ

mattclay committed:
Fix issues with docker_stack test:

- Leave swarm at end of test to avoid breaking docker_swarm test.
- Do not update stack to avoid error leaving swarm.
- Put test back in correct group.

joaopsys committed:
Add input validation for the new asn parameter

Fix asn type from str to int

joaopsys committed:
Remove whitespaces

dagwieers committed:
A few small fixes

felixfontein committed:
integration tests for docker_container (#45747)

* First round of integration tests for docker_container.

* Added regression test for #45700.

* Work around dict order randomization.

CastawayEGR committed:
VMware: vCenter plugin : new module (#44963)

jdufresne committed:
Correct capitalization of PyPI (#45698)

As spelled on https://pypi.org/.

dagwieers committed:
Docs: Avoid use of 'default: null' (#45795)

Various modules document the default 'null' value, but it causes None to
be shown in the documentation explicitly.

trishnaguha committed:
fix nxos_facts indefinite hang for text based output (#45845)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

pilou- committed:
doc: exercising module code locally: update cmd (#45697)

The documented way to execute module code locally wasn't always working:

    $ python ./lib/ansible/modules/files/file.py <<< '{"ANSIBLE_MODULE_ARGS": {}}'
    Traceback (most recent call last):
      File "./lib/ansible/modules/files/file.py", line 177, in <module>
        from ansible.module_utils.basic import AnsibleModule
      File "~/ansible/lib/ansible/module_utils/basic.py", line 78, in <module>
        import tempfile
      File "~/ansible/lib/ansible/modules/files/tempfile.py", line 69, in <module>
        from tempfile import mkstemp, mkdtemp
    ImportError: cannot import name 'mkstemp'

cordones committed:
incorrect sample eos_config usage (#45750)

no such option "config" for diff_against argument.
Further below, there is an example of eos_config using diff_against with a valid option.
+label: docsite_pr

mscherer committed:
Remove unused import and lines, to make linter happy (#45665)

Commit:
Update short description for NetApp E-Series modules. (#45820)

Update short description according to new standards

DSpeichert committed:
Migrate from MySQLdb to PyMySQL (#40123)

* Migrate from MySQLdb to PyMySQL

* Deduplicate driver loading and failure message

* Explain requirements

* Apply requirements docs change to proxysql too

* Add changelog

felixfontein committed:
Fixing HTTPError case of fetch_url for Python 3 compatibility. (#45628)

* Fixing HTTPError case of fetch_url for Python 3 compatibility.

* Adding unit test.

* PEP8.

* Changelog.

ehelms committed:
Deprecate foreman and katello modules (#42043)

* Deprecate foreman and katello modules in 2.8, remove in 2.12

Commit:
Added documentation for collectstatic --clear. (#42007)

* Added documentation for collectstatic --clear.

Added a description of the --clear argument used with the Django management
command, collectstatic. (When #1810 added this feature the documentation
was not updated).

aioue committed:
Update postgresql_user.py (#45777)

<!--- Your description here -->
Add expires example.
+label: docsite_pr

Commit:
Explain disk size in mount and add example (#41681)

* Explain disk size in mount and add example

- Explain mount with 8GB
- Add new example to start container with mount

* Removed trailing whitespace

carchi8py committed:
NetApp Bulk Doc changes (#45856)

* Doc changes

* fix ansibot issue

* Fix spacing

maxamillion committed:
handle yum and dnf lockfiles - fixes #44120 (#45359)

* handle yum and dnf lockfiles - fixes #44120

Signed-off-by: Adam Miller <admiller@redhat.com>

* fix logic problem to properly check for dnf lockfile glob

Signed-off-by: Adam Miller <admiller@redhat.com>

david-resnick667 committed:
datadog_monitor: fix template vars in 'name' and 'escalation_message' (#38483)

paulczar committed:
Fix bugs in softlayer dynamic inventory (#28178)

* --host hostname did not work due to calling wrong function

* softlayer api returns a bunch of extra stuff in the tagReference dict that
  makes --list output crazy long, like over a terminal buffer for just one server
  this culls out the extranneous information and only inserts the actual user
  provided tags to each server.

jbenden committed:
New module: Xfconf-based management of the Xfce 4 DE (#28112)

ansibot committed:
Fixing but on version check when the "Apache/2.4.x (Distro)" regex is not met (#27457)

jamiesonio committed:
Add environmental variables to allow for the script to be run without consul_io.ini. This allows it to run dynamically without any prerequisite config file (#30900)

marc-sensenich committed:
Add Check Mode to consul_kv (#28934)

jborean93 committed:
go back to FreeBSD 11.1 for tests due to 11.2 stability issues (#45872)

Commit:
adding os disk name (#45126)

* adding os disk name

* fixed test

Commit:
Fixing assigning ssl certificate to http listener in app gateway (#45830)

* fixed ssl certificate reference

* modified test

dagwieers committed:
Fix incorrect type in arg_spec

jborean93 committed:
append AWS region to test instance file (#45877)

mattclay committed:
Make ansible-test available in the bin directory. (#45876)

dagwieers committed:
Add missing ConnectionError imports (#45879)

Small fix

dagwieers committed:
Fix pylint error on utf8 character in string (#45880)

Small fix

mattclay committed:
Support comments in ansible-test flat files.

Commit:
Fixing azure_rm_containerregistry_facts (#45897)

* fixing cr facts

* final fix?

* fixed

* syntax

* sanity

jborean93 committed:
async: use async_dir for the async results file directory (#45461)

* win async: use async_dir for the async results file directory

* tried to unify POSIX and PowerShell async implementations of async_dir

* fix sanity issue

zeenlym committed:
ec2_vpc_igw: migrate to boto3 (#45346)

* ec2_vpc_igw: draft migrate to boto3

* ec2_vpc_igw: migrate to boto3

* ec2_vpc_igw: fix error and implement code review

* ec2_vpc_igw: always catch BotoCoreError with ClientError, remove unused fail method

* ec2_vpc_igw: fix pep error

acozine committed:
Explain what happens when gpgcheck is not set in yum

PR #45796

gundalow committed:
BOTMETA - redfish (#45909)

Qalthos committed:
Move persistent connections to only use registered variables (#45616)

* Try to intuit proper plugins to send to ansible-connection

* Move sub-plugins to init so that vars will be populated in executor

* Fix connection unit tests

max-allan-surevine committed:
Update keycloak_client.py (#43547)

* Update keycloak_client.py to document the 'realm' parameter, eliminate E322 ignore

maxamillion committed:
fix dnf wildcard pkg removal - fixes #27744 and #36970 (#45357)

Signed-off-by: Adam Miller <admiller@redhat.com>

remyleone committed:
Add an option to enable public ip at server creation (#44826)

larry-s committed:
The keyword 'cloud' has been deprecated in favor of the 'profile' keyword by os-client-config. (#31389)

decadent committed:
Clarified documentation for the unsafe_writes option (#27471)

* Clarified documentation for the unsafe_writes option per #24449.

remyleone committed:
Add Scaleway co-worker (#45925)

dagwieers committed:
Fix incorrect use of subprocess.CalledProcessError (#45890)

sivel committed:
Ensure that the src file contents is converted to unicode in diff info (#45744)

* Ensure that the src file contents is converted to unicode in diff info. Fixes #45717

* Fix up and cleanup

* The diff functionality in the callback plugins should have the
  to_text() calls removed since we're now doing it in ActionBase
* catching of UnicodeError and warnings in the callback diff
  functionality from 61d01f549f2143fd9adfa4ffae42f09d24649c26 haven't been
  needed since we switched to to_text so remove them.
* Add a note to ActionBase's diff function giving an example of when the
  diff function will be inaccurate and how to fix it

* Fix callback get_diff() tests

I believe the unittests of callback's get_diff() were wrong.  They were
sending in a list where strings were expected.  Because previous code
was transforming the lists into strings via their repr, the previous
tests did not fail but they would have formatted the test cases output
in an odd way if we had looked at it.

isopy committed:
Update azure_rm_virtualmachine.py (#45325)

* Update azure_rm_virtualmachine.py

A lot of Azure images are not cloud-init ready and need to be prepared manually before attempting to use the custom_data option of this module. Adding a line to the description to make others aware. If, like me, they are used to working with AWS AMIs that all seem to have cloud-init baked in, this could prevent some troubleshooting as to why their custom_data scripts aren't running in Azure.

* fixing addition to azure_rm_virtualmachine.py

* Update azure_rm_virtualmachine.py

* final update

* fixed url syntax

* nuked trailing whitespace

agaffney committed:
Add number of skipped tasks to play recap in 'default' callback (#45797)

Qalthos committed:
Clean up after ansible-connection if failure occurred in start() (#45929)

lwm committed:
Linode: docs note for py3 limitation and deprecated API (#44723)

* Add note that linode-python having no py3 support.

Closes https://github.com/ansible/ansible/issues/44695.

* Add note that the module uses the deprecated API.

lwm committed:
Use dashes not underscores for linode labels. (#44719)

This is because underscores are illegals in hostnames and users might
want to create the same host names are the name of the Linode machine
(as in the report in #30059).

Closes https://github.com/ansible/ansible/issues/30059.

samdoran committed:
Remove temp checksum file (#45871)

Spredzy committed:
inventory/tower: authors -< author so Doc can pick it up (#45936)

jborean93 committed:
powershell - fix for ANSIBLE_KEEP_REMOTE_FILES on older Pythons (#45942)

mattclay committed:
Improve ansible-test match error handling.

jborean93 committed:
powershell: do not quote join_path result to reflect ShellBase join_path (#45944)

jborean93 committed:
add skip/docker alias to skip tests when running on Docker (#45955)

* add skip/docker alias to skip tests when running on Docker

* changed warning message wording

mattclay committed:
Improve error handling for docs-build test.

Akasurde committed:
VMware: new module vcenter_extension_facts (#45472)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

tstoner committed:
Changed when condition from checking image version to checking keys being (#45943)

defined

OMMHOA committed:
update example to YAML style in documentation (#45968)

The example is in a key=value format which is deprecated.

+label: docsite_pr

Akasurde committed:
Misc typo fixes (#45969)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

smolz committed:
Add docs for say.py to indicate how to list voices (#37737)

* Update say.py

* Update say.py

* Update say.py

colin-mccarthy committed:
[network] fix typo in ftd_configuration - all operation(s) are performed over REST API (#45977)

remyleone committed:
Add documentation about the query filters (#45835)

dagwieers committed:
Fix calling deprecate with correct arguments (#44726)

This fixes #44702

remyleone committed:
Add a guide for using Online with Ansible (#45277)

mattclay committed:
Add python.py coverage injector for ansible-test.

This can be used to run Python scripts from the repository with the
correct interpreter and allow collection of code coverage.

Useful for testing contrib inventory scripts.

samdoran committed:
Unify terms and UI between 1Password lookups and facts module (#45427)

* Unify login behavior between 1Password lookup plugins and module

- Use the same names for all credential aspects
- Only require the minimal amount of information for each
- Add more examples

* Change parameter terms

- use terms in line with 1Password documentation.
- update examples
- update tests

* Improve error messages in lookup plugin

* Unify onepassword_facts with lookup plugins

- use same methods and logic for signing in or reusing existing session
- unify terms with lookup plugins

* Change rc test for determing login

An rc other than 1 can be returned when a current login session does not exist.

* Create AnsibleModuleError class

ansible.errors is not available to modules, so create an AnsibleModuleError class within the module

Do not user os.path.expanduser since this is already done by virtue of the type being "path" in the argument spec.

* Add note about risk with fact caching sensitive data

* Add note on op version that was used for testing

mattclay committed:
Bug fixes and cleanup for ansible-test. (#45991)

* Remove unused imports.
* Clean up ConfigParser usage in ansible-test.
* Fix bare except statements in ansible-test.
* Miscellaneous cleanup from PyCharm inspections.
* Enable pylint no-self-use for ansible-test.
* Remove obsolete pylint ignores for Python 3.7.
* Fix shellcheck issuers under newer shellcheck.
* Use newer path for ansible-test.
* Fix issues in code-smell tests.

bcoca committed:
added note about cli execution of include_x (#45227)

mattclay committed:
Fix integration test library search path.

This prevents tests from loading modules outside the source tree,
which could result in testing the wrong module if a system-wide
install is present, or custom modules exist.

a-harvie committed:
Add 'signalled' state to supervisorctl module with associated 'signal' parameter. (#32694)

https://github.com/ansible/ansible/issues/29847

mrmagooey committed:
Improve symlink install error message in setup.py

PR #45132

Co-authored-by: Sviatoslav Sydorenko <wk@sydorenko.org.ua>

smnmtzgr committed:
Update developing_modules_general_aci.rst

<!--- Your description here -->

+label: docsite_pr

mattclay committed:
Fix ansible-test docker python version handling.

This removes the old name based version detection behavior and
uses versions defined in the docker completion file instead, as
the new containers do not follow the old naming scheme.

mattclay committed:
Improve stability of Docker integration tests. (#46005)

nitzmahone committed:
fix premature exit when RG doesn't exist (#46013)

* fixes #45941
* corrects regression introduced by #26104; when the resource group doesn't exist, the module exits prematurely with an error instead of creating it.

mattclay committed:
Disable failing sts_assume_role integration test.

s-hertel committed:
Fix and re-enable sts_assume_role integration tests (#46026)

* Fix the STS assume role error message assertion when the role to assume does not exist.

Akasurde committed:
VMware: vmware_inventory contrib testing (#42879)

* VMware: vmware_inventory contrib testing
* Use python.py to run inventory script during test.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

tolland committed:
digital_ocean_domain: fix to allow updating the module param ip between calls (#42009)

* update the domain record with the new ip from params if resource exists

* fixes for pylint and pep8 checks

resmo committed:
cs_loadbalancer_rule_member: fix error handling (#46012)

* make use of query_api
* fix sanity

rndmh3ro committed:
Update yaml-style in password-lookup example (#46046)

* Update yaml-style in password-lookup example

##### SUMMARY

Update the  yaml-style in a password-lookup example to match best-practices.

##### ISSUE TYPE
- Docs Pull Request


##### COMPONENT NAME
password_lookup plugin

##### ANSIBLE VERSION
devel

* remove whitespace

felixfontein committed:
docker_container: fix various idempotency problems and non-working options (#45905)

* Sorting args.

* Doing comparisons of options with container parameters in a more context-sensitive way.

This prevents unnecessary restarts, or missing restarts (f.ex. if parameters are removed from ``cmd``).

* Make blkio_weight work.

* Fix cap_drop idempotency problem.

* Making groups idempotent if it contains integers.

* Make cpuset_mems work.

* Make dns_opts work.

* Fixing log_opts: docker expects string values, returns error for integer.

* Adding tests from felixfontein/ansible-docker_container-test#2.

* Make uts work.

* Adding changelog entry.

* Forgot option security_opts.

* Fixing typo.

* Explain strict set(dict) comparison a bit more.

* Improving idempotency tests.

* Making dns_servers a list, since the ordering is relevant.

* Making dns_search_domains a list, since the ordering is relevant.

* Improving dns_search_domains/dns_servers.

* Fixing entrypoint test.

* Making sure options are only supported for correct docker-py versions.

rhodain committed:
Spelling corrections

rhodain committed:
Spelling corrections

rhodain committed:
Merge branch 'ovirt_disk_port_integer' into ovirt_disk_refresh_lun_disk

rhodain committed:
Merge branch 'ovirt_disk_refresh_lun_disk' of https://github.com/rhodain/ansible into ovirt_disk_refresh_lun_disk

Qalthos committed:
Try to be more helpful when JSON gives up (#45600)

* Try to be more helpful when JSON gives up

rhodain committed:
Merge branch 'ovirt_disk_refresh_lun_disk' of https://github.com/rhodain/ansible into ovirt_disk_refresh_lun_disk

gundalow commented:
Hi if you have a look at http://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html that should help you with rebasing this PR
